eng/build.ps1

```diff
@@ -66,6 +66,11 @@ if ($PSVersionTable.PSVersion.Major -lt "5") {
     exit 1
 }
 
+$regKeyProperty = Get-ItemProperty -Path HKLM:\SYSTEM\CurrentControlSet\Control\FileSystem -Name "LongPathsEnabled" -ErrorAction Ignore
```

Comment:
This isn't the right place for the check. It should be inside the main `try / finally` block. When it is outside like this the script is not guaranteed to have proper exit codes. If any of our helpers throw an exception, which they are designed to do in the face of error, the script will fail but exit with code 0 indicating success.

Reply:
Here we ignore any exceptions, so this will never fail.

Reply:
Even if "this will never fail" was true this is still the wrong place for the check. Powershell error management is really hard to get right. Asking future authors who change this line of code to correctly understand the implication of the change is not realistic. Putting it in the `try / finally` though ensures that it feeds into all of the safety checks we've put into place here. 

BTW: this can indeed fail. `$WarningPreference = "Stop"`

Reply:
Ok, ok. Let's move this to the try block.

Reply:
And the Powershell version check too?

Reply:
K, I'll put up a PR with `try/catch`